core/src/main/scala/org/bitcoins/core/crypto/bip32/BIP32Path.scala

```diff
@@ -0,0 +1,70 @@
+package org.bitcoins.core.crypto.bip32
+
+import org.bitcoins.core.number.UInt32
+
+abstract class BIP32Path {
+  def children: Vector[BIP32Child]
+
+  override def toString: String =
+    children
+      .map {
+        case BIP32Child(index, hardened) =>
+          index.toString + (if (hardened) "'" else "")
+      }
+      .fold("m")((accum, curr) => accum + "/" + curr)
+
+}
+
+object BIP32Path {
+
+  /**
+    * The empty BIP32 path "m"
+    */
+  def empty: BIP32Path = BIP32PathImpl(Vector.empty)
+
+  private case class BIP32PathImpl(children: Vector[BIP32Child])
+      extends BIP32Path
+
+  def apply(children: Vector[BIP32Child]): BIP32Path = BIP32PathImpl(children)
+
+  def fromString(string: String): BIP32Path = {
+    val parts = string
+      .split("/")
+      .toVector
+      .map(_.trim)
```

Comment:
why did you add in `.trim` and `.toLowerCase`? Is this in the BIP somewhere that this is specified? 

If not, is there any way this can cause incompatabilities where users of the wallet infrastructure are storing _invalid_ strings some where that we are deriving seemingly valid keys? 

I think we need to be careful with sanitizing input, it is probably best in situations like this to throw. Can you link to any other implementations that do this as well?

Reply:
You're right about upper/lower case, removed that. There are many examples of people having/lacking whitespace between their segments. Linked to one example for each in comment above.